client/src/app/shared/components/FilterToolbar/FilterToolbar.tsx

```diff
@@ -113,6 +117,34 @@ export const FilterToolbar = <TItem, TFilterCategoryKey extends string>({
     (category) => category.key === currenTFilterCategoryKey
   );
 
+  const groupedDropdownItems = filterGroups?.map((filterGroup) => (
+    <DropdownGroup label={filterGroup} key={filterGroup}>
+      {filterCategories
+        .filter((filterCategory) => filterCategory.filterGroup === filterGroup)
+        .map((filterCategory) => {
+          return (
+            <DropdownItem
+              id={`filter-category-${filterCategory.key}`}
+              key={filterCategory.key}
+              onClick={() => onCategorySelect(filterCategory)}
+            >
+              {filterCategory.title}
+            </DropdownItem>
```

Comment:
You could probably factor out this DropdownItem piece that repeats below as a `renderDropdownItem` function or its own component, if that's worth it. Maybe not worth it.